pkg/dns/provider/errors/errors.go

```diff
@@ -50,3 +50,20 @@ type NoSuchHostedZone struct {
 func (e *NoSuchHostedZone) Error() string {
 	return fmt.Sprintf("No such hosted zone %s: %s", e.ZoneId, e.Err)
 }
+
+func NewThrottlingError(err error) *ThrottlingError {
```

Comment:
Are there other providers suffering from throttling besides the AWS one?
Then these should be adjusted as well

Reply:
I'm not aware of throttling errors for other providers. Let's start with AWS now.